### DISCUSSION TITLE
Is There a Good Way to Run Bottle Rocket on Other Clouds?

### DISCUSSION THREAD

**Discussion by zicklag** [bottlerocket-os/bottlerocket#4097] [Q&A]
Hey there! I'm really interested in Bottle Rocket, but I'm unsure what the process should be for deploying to something like Linode or Digital Ocean?

Should I just build a bare metal image, and then try to deploy that? Is there anybody who's tried running Bottle Rocket on a non-AWS or VMWare cloud?

**Comment by yeazelm** [comment]
Hello @zicklag, thanks for opening up this discussion! We would love to have Bottlerocket running in all kinds of places and have done some work to enable that to happen. We do have open issues already for enabling a few other clouds (#968, #1114). These issues haven't seen much progress but we also have enabled [VMware images](https://github.com/bottlerocket-os/bottlerocket/blob/develop/QUICKSTART-VMWARE.md) and [metal support](https://github.com/bottlerocket-os/bottlerocket/blob/develop/PROVISIONING-METAL.md) which might be useful places to get started. Using a metal or VMware image might be a decent place to start for places like Linode or Digital Ocean. Bootstrapping might involve some decent effort depending on how the specific provider treats node bring up since Bottlerocket doesn't have `cloud-init` but instead its own user-data TOML definitions and logic to do node initialization. If there is a guide for bringing a VMware image to the specific place you are wanting to run Bottlerocket, that is where I would start since the guides might be pretty close to what you need. 

One place of note is network configuration, [the metal doc](https://github.com/bottlerocket-os/bottlerocket/blob/develop/PROVISIONING-METAL.md#network-interface-configuration) has some good data on using a `net.toml` to configure networking which will be needed if using the metal image. 

We would love to hear how things progress and we’re happy to provide guidance on what build system changes might be required if you run into issues!

**Comment by larvacea** [comment]
We have restructured Bottlerocket to make it simpler to build your own customized variant. Our `bottlerocket-core-kit` repository contains both our first-party code and the packages we build for use in our own variants, and this repo defines those variants and pulls components as needed from the core kit. This is very much a work in progress, but it may simplify the process of creating your own non-AWS variant.

**Reply by zrml** [reply]
Any news on this effort and how AWS sees this project evolving for porting across Azure, GCP and on-prem? Thanks
